src/api/v1/users/schema.ts

```diff
@@ -16,8 +16,9 @@ export const userName =
   Joi.string()
     .min(3)
     .max(100)
-    .regex(/[^\w\s\d]/, { name: 'alphanumeric', invert: true })
-    .lowercase();
+    .regex(/^[a-zA-Z]{2,}\s?[a-zA-z]*['\-]?[a-zA-Z]*['\- ]?([a-zA-Z]{1,})?/, { name: 'alpha' })
```

Comment:
Rather than come up with ever more specific regexes to handle every edge case you can think of, why not just use a permissive regex so you don't need to worry about the edge cases you have inevitably missed.

There is really no need to tightly constrain user names. TBH, this is why I wasn't really a fan of lowercasing names in the database. It's purely a developer convenience.

I'm not going to say we should deal with this now though. We should ultimately deal with it by using the `citext` postgres module or a functional index to support case insensitive searches.

No action.